src/app/Library/CrudPanel/Traits/FieldsProtectedMethods.php

```diff
@@ -169,20 +169,25 @@ protected function makeSureFieldHasRelationshipData($field)
         return $field;
     }
 
-    protected function makeSureFieldNameMatchesRelation($field)
+    protected function overwriteFieldNameFromEntity($field)
     {
-        switch ($field['relation_type']) {
-            case 'BelongsTo':
-                if (count(explode('.', $field['entity'])) == count(explode('.', $this->getOnlyRelationEntity($field)))) {
-                    $field['name'] = implode('.', array_slice(explode('.', $field['entity']), 0, -1));
-                    $relation = $this->getRelationInstance($field);
-                    if (! empty($field['name'])) {
-                        $field['name'] .= '.';
-                    }
-                    $field['name'] .= $relation->getForeignKeyName();
-                }
-
-        break;
+        // if the entity doesn't have a dot, it means we don't need to overwrite the name
+        if (! Str::contains($field['entity'], '.')) {
+            return $field;
+        }
```

Comment:
When is this method useful @pxpm ? The changes I made were thinking that it's only useful when entity has become dot notation. For 1-1 relations. Am I wrong?

Reply:
While we're at it, can you please explain (again... I know), when the `entity` can use dot notation? An example please? Was this possible in 4.0 too? I personally have never used dot notation inside entity. Inside name, maybe. But inside entity never.

Thanks!

Reply:
Entity with dot notation was added for 1-1 relations. We are in monster, hasOne address and address has a relation BelongsTo Country (where you can selected the country). In this scenario the field would be:

```php
$this->crud->addField('address.country');
```
Before your changes that would translate to:

```
[
'name' => address[country_id]
'entity' => address.country
....
]
```
Now i am pretty sure it will not work, because technically the relation from address to country is a belongsTo. 

I am pretty sure we don't need to overwrite in case name does not have dot notation, but in case field has dot notation it should be overriden.

Reply:
Ok, that's great, that's what it does now. The `address.country` field in Monsters that tests this use case seems to work fine, so I think we're good.